dm/syncer/data_validator.go

```diff
@@ -777,3 +782,33 @@ func genColData(v interface{}) string {
 	s := fmt.Sprintf("%v", v)
 	return s
 }
+
+func (v *DataValidator) GetValidationStatus() []*pb.ValidationStatus {
+	failpoint.Inject("MockValidationQuery", func() {
+		failpoint.Return([]*pb.ValidationStatus{
+			{
+				SrcTable:         "`testdb1`.`testtable1`",
+				DstTable:         "`dstdb`.`dsttable`",
+				ValidationStatus: pb.Stage_Running.String(),
+			},
+			{
+				SrcTable:         "`testdb2`.`testtable2`",
+				DstTable:         "`dstdb`.`dsttable`",
+				ValidationStatus: pb.Stage_Stopped.String(),
+				Message:          tableWithoutPrimaryKeyMsg,
+			},
+		})
+	})
+	v.RLock()
```

Comment:
we need to protect all access to `v.tableStatus` using lock. in `processRowsEvent` we're writing to it without lock, may cause data race.

Reply:
I've tried to fix it by requiring exclusive lock before processing rows event. PTAL again.

Reply:
it's ok for now, i'm refactoring `processRowsEvent` to reuse `pkg/sqlmodel`, will try to make the lock range smaller.